### DISCUSSION TITLE
Revset: visible version of hidden commit

### DISCUSSION THREAD

**Discussion by crepererum** [jj-vcs/jj#4263] [Q&A] [answered]
Is there are way to transform a revset that may or may not contain hidden commits to their visible versions? E.g. if you run `mine() & tracked_remote_branches()` you may end up with hidden commits because the local version of the change might already be rebased:

![image](https://github.com/user-attachments/assets/bd40d376-f43c-4877-a2a7-6da364faffa9)

![image](https://github.com/user-attachments/assets/ba8c3220-d81c-4695-aac6-3b62fb0ed252)

I've tried to use `descendants` but this only resulted in an empty result:

![image](https://github.com/user-attachments/assets/bcde7fe1-efb9-4cb5-a0b5-e3a7fb20da15)

**Comment by bnjmnt4n** [comment, marked as answer]
There's no existing revset functions which can do this. The tracking issue for this is https://github.com/martinvonz/jj/issues/4129.
